### ISSUE TITLE
NPE in LogicalOggStreamImpl due to wrong magic number in audio asset

### ISSUE THREAD

**Issue by stephengold** [jMonkeyEngine/jmonkeyengine#1244]
An issue was raised at the Forum
    https://hub.jmonkeyengine.org/t/solved-audio-cannot-be-play-in-appstate/42712

Loading a non-OGG audio file with `com.jme3.audio.plugins.OGGLoader` caused a `NullPointerException` to be thrown in `LogicalOggStreamImpl.getNextOggPacket()`. Why wasn't the error handled by jme3-jogg?

Perhaps due to a commented-out null check in OGGLoader:
    https://github.com/jMonkeyEngine/jmonkeyengine/blob/master/jme3-jogg/src/main/java/com/jme3/audio/plugins/OGGLoader.java#L272

**Comment by pspeed42** [comment]
What's interesting is that for all intents and purposes, it has always been that way.  That code was commented out even when jme3 was moved to the svn trunk 9 years ago.  I'm unable to trace farther back than that but I suspect it was inherited even from jme2.  

It's like archaeology.

**Comment by stephengold** [comment]
I reproduced the NPE using "TestOgg.java" and a corrupted version of "Foot steps.ogg". To my surprise, `loStream` is non-null. So un-commenting the null check won't solve this issue.

The best place to resolve this issue would be in the "j-ogg-all" library itself. We're using v1.0.0 of j-ogg-all, which dates from 2014. I searched the web and didn't find a newer version. Neither of the websites mentioned in the POM (http://j-ogg.de and http://svn.wever.org) exist anymore. Nor did I find the project at GitHub.

I extracted the source files from the sources JAR at MavenCentral and built a new class JAR with little difficulty. (There's a compile dependency on `'javax.media:jmf:2.1.1e'`.)

The NPE occurs because `currentPage` is `null` in the *second* `if` clause below:
```java
      if(currentPage==null) {
         currentPage=getNextOggPage();
      }

      do {
         if(currentSegmentIndex>=currentPage.getSegmentOffsets().length) {
```

It should be possible to build a modified j-ogg-all library that tests for `null` and throws a different exception in that case.

Alternatively, uncommenting the `throw` in line 231 of OggJmfPage.java might resolve this issue. However it might also cause usable files to be rejected:
```java
         if(capture!=0x5367674f) {
            //throw new FormatException("Ogg page does not start with 'OggS' (0x4f676753)");

            /*
            ** This condition is IMHO an error, but older Ogg files often contain
            ** pages with a different capture than OggS. I am not sure how to
            ** manage these pages, but the decoder seems to work properly, if
            ** the incorrect capture is simply ignored.
            */
```

Licensing won't be an issue. I obtained a snapshot of the license text (via The Wayback Machine), and here is what it said:
```text
I'm not a fan of legal nonsence, so I'll try to keep this short :) You are free to use, modify, resdistribute or include this software in your own free or commercial software. The only restriction is, that you make it obvious that your software is based on J-Ogg by including this notice in the documentation, about box or whereever you feel apropriate:

"This software is based on or using the J-Ogg library available from http://www.j-ogg.de and copyrighted by Tor-Einar Jarnbjo." 
```

However, I'm not sure we want to maintain our own version of a 7 year-old library...

**Comment by stephengold** [comment]
Fixed in "master" branch at 6d31571 .
